components/renku_data_services/storage_adapters/core.py

```diff
@@ -68,7 +90,19 @@ async def update_storage(self, storage_id: str, **kwargs) -> models.CloudStorage
 
                 if storage is None:
                     raise errors.MissingResourceError(message=f"The storage with id '{storage_id}' cannot be found")
+                if storage[0].project_id != user.project_id:
```

Comment:
Along with my comment about the GitlabAuthenticator it would be nice to hide the authorization complexity in the Authenticator and not see it here. The Authenticator or a decorator on an endpoint handler can raise this error if things do not match.